flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,719 @@
+    'movie/bluray': 2050,
+    'movie/3d': 2060,
+    'movie/dvd': 2070,
+    'movie/web-dl': 2080,  # unofficial
+    # Music
+    'music': 3000,
+    'music/mp3': 3010,
+    'music/video': 3020,
+    'music/audiobook': 3030,
+    'music/lossless': 3040,
+    'music/foreign': 3060,  # unofficial
+    # PC (Games + Software)
+    'pc': 4000,
+    'pc/0day': 4010,
+    'pc/iso': 4020,
+    'pc/mac': 4030,
+    'pc/mobile-other': 4040,
+    'pc/games': 4050,
+    'pc/mobile-ios': 4060,
+    'pc/mobile-android': 4070,
+    # TV
+    'tv': 5000,
+    'tv/web-dl': 5010,  # unofficial
+    'tv/foreign': 5020,
+    'tv/sd': 5030,
+    'tv/hd': 5040,
+    'tv/uhd': 5045,  # unofficial
+    'tv/other': 5050,
+    'tv/sport': 5060,
+    'tv/anime': 5070,
+    'tv/documentary': 5080,  # unofficial
+    # XXX
+    'xxx': 6000,
+    'xxx/dvd': 6010,
+    'xxx/wmv': 6020,
+    'xxx/xvid': 6030,
+    'xxx/x264': 6040,
+    'xxx/uhd': 6045,  # unofficial
+    'xxx/pack': 6050,
+    'xxx/imageset': 6060,
+    'xxx/packs': 6070,  # unofficial
+    'xxx/sd': 6080,  # unofficial
+    'xxx/web-dl': 6090,  # unofficial
+    # Misc
+    'misc': 7000,
+    'misc/misc': 7010,
+    'misc/ebook': 7020,
+    'misc/comics': 7030,
+    # Unknown
+    'not-determined': 7900
+}
+
+# NOTE: all lowercase only
+NAMESPACE_ATTRIBUTE_MAP = {
+    'grabs': int,
+    'size': int,
+    'files': int,
+    'usenetdate': datetime,
+    'password': bool,
+    'guid': basestring,
+    'hydraindexername': basestring,
+    'hydraindexerhost': basestring,
+    'hydraindexerscore': int
+}
+
+PLUGIN_LOOKUP_MAP = {
+    'tvdb': 'thetvdb_lookup',
+    'trakt': 'trakt_lookup',
+    'tvmaze': 'tvmaze_lookup',
+    'imdb': 'imdb_lookup',
+    'tmdb': 'tmdb_lookup'
+}
+
+NAMESPACE_NAME = 'newznab'
+NAMESPACE_URL = 'http://www.newznab.com/DTD/2010/feeds/attributes/'
+NAMESPACE_TAGNAME = 'attr'
+NAMESPACE_PREFIX = 'newznab_'
+ENCLOSURE_TYPE = 'application/x-nzb'
+
+
+# utils
+def list_insert_unique(in_list, idx, element, caseinsensitive=False):
+    if not isinstance(in_list, list) or element is None:
+        return
+    if isinstance(element, basestring):
+        if element.isspace() or element in in_list:
+            return
+        elif caseinsensitive is True and element.lower() in in_list:
+            return
+    elif element in in_list:
+        return
+    in_list.insert(idx, element)
+
+
+def list_append_unique(in_list, element, caseinsensitive=False):
+    list_insert_unique(in_list, len(in_list), element, caseinsensitive=caseinsensitive)
+
+
+def list_combine_unique(list_dst, list_source, caseinsensitive=False):
+    if not isinstance(list_dst, list) or not isinstance(list_source, list):
+        log.error('Cant combine lists, src/dst is not a listtype: %s %s' % (list_dst, list_source))
+    for item in list_source:
+        list_append_unique(list_dst, item, caseinsensitive=caseinsensitive)
+
+
+def safe_get(in_object, keynames, types=basestring, default=None, caseinsensitive=True):
```

Comment:
Well, for one, I'm not exactly sure what it's trying to do, or why. From the looks of the way this function is being used, it could be replaced with `if all(field in entry for field in ['tvdb_id', 'other', fields]):` We use duck typing extensively in flexget (and python in general,) if some plugin is filling these fields with data types that other plugins aren't expecting, we sortof want everything to fail and fix the issue rather than everything silently degrading.